lib/internal/http2/core.js

```diff
@@ -2430,15 +2436,19 @@ function connect(authority, options, listener) {
   const host = authority.hostname || authority.host || 'localhost';
 
   let socket;
-  switch (protocol) {
-    case 'http:':
-      socket = net.connect(port, host);
-      break;
-    case 'https:':
-      socket = tls.connect(port, host, initializeTLSOptions(options, host));
-      break;
-    default:
-      throw new errors.Error('ERR_HTTP2_UNSUPPORTED_PROTOCOL', protocol);
+  if (typeof options.createConnection === 'function') {
+    socket = options.createConnection(authority, options);
```

Comment:
should likely type check the return value

Reply:
I think all we could do here is doing some duck-type checking, e.g. verifying that `.write()` and `.on()` are present…

Reply:
Do we really need to? This is very much opt-in... What does `http` do for `createConnection`?

Reply:
Do we really need to? This is very much opt-in... What does `http` do for `createConnection`?

Reply:
It doesn’t do any explicit typechecking either, yes.

Reply:
Ok. I won't block on that. I much prefer the APIs to be stricter here and not gate based on what we do in http1 but I can live with this.